int-set/src/lib.rs

```diff
@@ -498,6 +552,79 @@ where
     }
 }
 
+struct IterAfter<SetIter, AllValuesIter>
+where
+    SetIter: Iterator<Item = u32>,
+    AllValuesIter: Iterator<Item = u32>,
+{
```

Comment:
this is a kind of funny point, but there's a general rust idiom towards separating data and implementation, and specifically around avoiding trait bounds on data types unless necessary, saving them for the actual impl blocks.

This both saves typing but also avoids the possible 'bounds contagion' where any type that contains another type that has bounds has to also declare the appropriate bounds.

This isn't particularly relevant here, but I think it's a good habit to avoid the bounds on the type decls where possible.

Reply:
Thanks, used this approach to merge IterAfter into Iter.